app/src/main/java/org/simple/clinic/bp/entry/confirmremovebloodpressure/ConfirmRemoveBloodPressureDialogController.kt

```diff
@@ -0,0 +1,43 @@
+package org.simple.clinic.bp.entry.confirmremovebloodpressure
+
+import io.reactivex.Observable
+import io.reactivex.ObservableTransformer
+import io.reactivex.rxkotlin.ofType
+import io.reactivex.rxkotlin.withLatestFrom
+import org.simple.clinic.ReplayUntilScreenIsDestroyed
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.bp.BloodPressureRepository
+import org.simple.clinic.bp.entry.ConfirmRemoveBloodPressureDialog
+import org.simple.clinic.widgets.UiEvent
+import javax.inject.Inject
+
+typealias Ui = ConfirmRemoveBloodPressureDialog
+typealias UiChange = (Ui) -> Unit
+
+class ConfirmRemoveBloodPressureDialogController @Inject constructor(
+    private val bloodPressureRepository: BloodPressureRepository
+) : ObservableTransformer<UiEvent, UiChange> {
+
+  override fun apply(events: Observable<UiEvent>): Observable<UiChange> {
+    val replayedEvents = ReplayUntilScreenIsDestroyed(events)
+        .compose(ReportAnalyticsEvents())
+        .replay()
+
+    return markBloodPressureAsDeleted(replayedEvents)
+  }
+
+  private fun markBloodPressureAsDeleted(events: Observable<UiEvent>): Observable<UiChange> {
+    val savedBloodPressureMeasurementStream = events
+        .ofType<ConfirmRemoveBloodPressureDialogCreated>()
+        .flatMapSingle { bloodPressureRepository.measurement(it.bloodPressureMeasurementUuid) }
+
```

Comment:
Feel free to use `Bp` instead of `BloodPressure` in your code. We're okay with using acronyms.

Reply:
It's alright, I'm not a huge fan of acronyms. 😌